api/src/main/java/com/cloud/deploy/DeploymentPlanner.java

```diff
@@ -104,13 +107,26 @@ public ExcludeList(Set<Long> dcIds, Set<Long> podIds, Set<Long> clusterIds, Set<
             }
         }
 
+        private void logAvoid(Class<?> scope, CloudException e) {
+            Long id = null;
+            if (e instanceof InsufficientCapacityException) {
+                id = ((InsufficientCapacityException) e).getId();
+            } else if (e instanceof ResourceUnavailableException) {
+                id = ((ResourceUnavailableException) e).getResourceId();
+            } else {
+                LOGGER.trace(String.format("Failed to log avoided component due to unexpected exception type [%s].", e.getMessage()));
+                return;
+            }
+            LOGGER.trace(String.format("Adding %s [%s] to the avoid set due to [%s].", scope.getSimpleName(), id, e.getMessage()));
```

Comment:
should this be debug or even info?

Reply:
Changed to debug, thanks for the suggestion.